### PR TITLE
gh-2027 Add equals to Children of ToBytesSerialiser

### PR DESCRIPTION
**Pull request by EmekeN**
Implementing Equals and Hashcode methods of children of ToBytesSerialiser. Addresses issue #2027

### CODE REVIEW

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java [hunk 1]
```diff
@@ -44,19 +48,32 @@ public boolean canHandle(final Class clazz) {
...
+                    throw new SerialisationException("Was not able to process EntrySet of Map");
+                }
             }
         } catch (final IOException e) {
             throw new SerialisationException(e.getMessage(), e);
         }
         return builder.toArray();
     }
 
+    protected void checkSerialiers(final ToBytesSerialiser keySerialiser, final ToBytesSerialiser valueSerialiser) {
```
Comment (d47853) [inline core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java]:
```suggestion
    protected void checkSerialisers(final ToBytesSerialiser keySerialiser, final ToBytesSerialiser valueSerialiser) {
```

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java [hunk 2]
```diff
@@ -72,8 +89,11 @@ public boolean canHandle(final Class clazz) {
         final int arrayLength = bytes.length;
         int carriage = 0;
         while (carriage < arrayLength) {
-            LengthValueBytesSerialiserUtil.ObjectCarriage c = LengthValueBytesSerialiserUtil.deserialiseNextObject(getKeySerialiser(), carriage, bytes);
-            LengthValueBytesSerialiserUtil.ObjectCarriage c2 = LengthValueBytesSerialiserUtil.deserialiseNextObject(getValueSerialiser(), c.getCarriage(), bytes);
+            final ToBytesSerialiser keySerialiser = getKeySerialiser();
+            final ToBytesSerialiser valueSerialiser = getValueSerialiser();
+            checkSerialiers(keySerialiser, valueSerialiser);
```
Comment (d47853) [inline core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java]:
```suggestion
            checkSerialisers(keySerialiser, valueSerialiser);
```

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java [hunk 3]
```diff
@@ -44,19 +48,32 @@ public boolean canHandle(final Class clazz) {
...
         try {
-            for (final Map.Entry entry : object.entrySet()) {
-                builder.appendLengthValueFromObjectToByteStream(getKeySerialiser(), entry.getKey());
-                builder.appendLengthValueFromObjectToByteStream(getValueSerialiser(), entry.getValue());
+            for (final Object o : object.entrySet()) {
+                if (o instanceof Map.Entry) {
+                    Map.Entry entry = (Map.Entry) o;
+                    final ToBytesSerialiser keySerialiser = getKeySerialiser();
+                    final ToBytesSerialiser valueSerialiser = getValueSerialiser();
+                    checkSerialiers(keySerialiser, valueSerialiser);
```
[the remark this answers is not in the document]
Comment (EmekeN) [reply core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java]: changed all references of checkSerialiers to checkSerialisers

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/ToBytesSerialiser.java [hunk 4]
```diff
@@ -73,10 +73,9 @@ default T deserialise(final byte[] allBytes, final int offset, final int length)
      * @param bytes the bytes to deserialise
      * @return T the deserialised object
      * @throws SerialisationException if the object fails to deserialise
-     *
+     * <p>
```
[the remark this answers is not in the document]
Comment (EmekeN) [reply core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/ToBytesSerialiser.java]: fixed

**Comment by GCHQDev404** [comment]
@EmekeN Thanks for you work on this.
The build is failing, Travis message: `There are 12 errors reported by Checkstyle`
Can you correct please, let me know if you need further assistance.

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (GCHQDev404) [reply core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/MapSerialiser.java]: can you confirm you pushed this change please.

**Comment by GCHQDev404** [comment]
@EmekeN thanks for your work on this.
Travis is reporting: `on project serialisation: Failed during checkstyle execution: There are 12 errors reported by Checkstyle`
Can you correct these errors please.
Ask if you need any help.

**Comment by GCHQDev404** [comment]
@EmekeN do you require assistance?

**Comment by EmekeN** [comment]
> @EmekeN do you require assistance?

No, I'll be able to  finish the PR I had to table the work for a while while more pressing things came up.

**Comment by GCHQDev404** [comment]
Inactive PR

**Pull request closed without merging**